src/components/Swap/SwapButton/index.tsx

```diff
@@ -67,80 +67,82 @@ export default memo(function SwapButton({ disabled }: SwapButtonProps) {
   // Close the review modal on chain change.
   useEffect(() => setOpen(false), [chainId])
 
-  const addTransaction = useAddTransaction()
+  const addTransactionInfo = useAddTransaction()
   const setDisplayTxHash = useUpdateAtom(displayTxHashAtom)
-  const setOldestValidBlock = useSetOldestValidBlock()
+  const [, setSwapAmount] = useSwapAmount(Field.INPUT)
+  const resetSwapAmount = useCallback(() => setSwapAmount(''), [setSwapAmount])
```

Comment:
I was looking for *where* the the `Field.INPUT` setting was happening, then realized it's probably happening implicitly here. Just want to point out that this feels a little hard to parse. I don't know if someone reading this code w/o context would be able to recognize that is happening.

Reply:
I thought that by semantically naming a derived setter I would make it easier to parse, so this is great feedback. I'll opt for the more explicit `setSwapAmount('')` inlined, and alias `setSwapAmount` to `setInputAmount` or similar.

Reply:
specifically I mean it's not immediately clear to me that these lines reset the widget to an `exactInput` swap if it was not immediately in that state. The derived setter makes sense for the `Amount` for sure, just not the `Field`/`tradeType`